esiclient/tests/functional/v1/test_node.py

```diff
@@ -137,3 +139,180 @@ def test_admin_can_node_power_on_off(self):
+                       'owner', self.projects['random']['id'])
+        self.addCleanup(utils.node_set,
+                        self.clients['admin'],
+                        self.node, 'owner')
+
+        utils.node_console_enable(self.clients['random-member'], self.node)
+        self.addCleanup(utils.node_console_disable,
+                        self.clients['admin'],
+                        self.node,
+                        fail_ok=True)
+        time.sleep(10)
+        console = utils.node_console_show(self.clients['random-member'],
+                                          self.node)
+        assert console['console_enabled']
+
+        utils.node_console_disable(self.clients['random-member'], self.node)
+        time.sleep(10)
+        console = utils.node_console_show(self.clients['random-member'],
+                                          self.node)
+        assert not console['console_enabled']
+
+    def test_lessee_can_enable_console(self):
+        """Tests lessee console functionality.
+
+        Tests that a lessee can perform the steps to setup an ipmitool-socat
+        serial console.
+
+        Test steps:
+            1) Set lessee of node to random project
+            2) Check that the lessee can run command
+                `baremetal node console enable <node>`
+            3) Check that lessee can run command
+                baremetal node console show <node>`
+            4) Check that lessee can run command
+                `baremetal node console disable <node>`
+            5) (cleanup) Unset the lessee
+            6) (cleanup) disable console
+        """
+
+        node = utils.node_show(self.clients['admin'], self.node)
+        if node['provision_state'] != 'manageable':
+            utils.node_set_provision_state(self.clients['admin'],
+                                           self.node, 'manage')
+
+        utils.node_set(self.clients['admin'],
+                       self.node,
+                       'lessee', self.projects['random']['id'])
+        self.addCleanup(utils.node_set,
+                        self.clients['admin'],
+                        self.node, 'lessee')
+
+        utils.node_console_enable(self.clients['random-member'], self.node)
+        self.addCleanup(utils.node_console_disable,
+                        self.clients['admin'],
+                        self.node,
+                        fail_ok=True)
+        time.sleep(10)
+        console = utils.node_console_show(self.clients['random-member'],
+                                          self.node)
+        assert console['console_enabled']
+
+        utils.node_console_disable(self.clients['random-member'], self.node)
+        time.sleep(10)
+        console = utils.node_console_show(self.clients['random-member'],
+                                          self.node)
+        assert not console['console_enabled']
+
+    def test_random_cannot_enable_console(self):
+        """Tests lessee console functionality.
```

Comment:
These comments may need to be updated

Reply:
fixed

Reply:
Ah, I think you may have forgotten to update this test description!